### ISSUE TITLE
Rubocop disable directives matching has odd behaviour

### ISSUE THREAD

**Issue by tejasbubane** [rubocop/rubocop#13965]
Rubocop disable directives seem to be matching patterns causing odd behaviour.

--------

## Expected behavior

Rubocop should report with unknown cop error.

## Actual behavior

Rubocop silently passes.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

a = 1 # rubocop:disable Lint/ selessAssignment
b = 2 # rubocop:disable Lint/lAssignment
```

Note the extra spacing and typo in the disable directives.

Rubocop does not report any offenses. Not sure what is happening here. Is it disabling the entire `Lint` department? Or is it guessing the cop name?

Change the name slightly (change `lAssignment` to `Assignment`) and it starts reporting offense!

```ruby
# frozen_string_literal: true

a = 1 # rubocop:disable Lint/ selessAssignment
b = 2 # rubocop:disable Lint/Assignment
```

```
Inspecting 1 file
W

Offenses:

test.rb:4:1: W: [Correctable] Lint/UselessAssignment: Useless assignment to variable - b.
b = 2 # rubocop:disable Lint/Assignment
^
test.rb:4:7: W: [Correctable] Lint/RedundantCopDisableDirective: Unnecessary disabling of Lint/Assignment (did you mean Lint/SelfAssignment?).
b = 2 # rubocop:disable Lint/Assignment
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

## RuboCop version

```
1.73.2 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 2.7, running on ruby 3.4.2) [arm64-darwin24]
```

**Comment by dvandersluis** [comment]
I opened #13972 which fixes the `b = 2 # rubocop:disable Lint/Assignment` example. 

The other example is very interesting and only applies when the directive disables a `Lint/` cop improperly. `a = 1 # rubocop:disable Lint/ selessAssignment` is processed as a disable of the *entire* `Lint` department. Although `Lint/CopDirectiveSyntax` would normally catch this case as a malformed disable directive, the offense is not reported because of the disabling of `Lint`. You can see how this should work if you do this with any other department:

```ruby
# test.rb

a = 1 # rubocop:disable Layout/ LineLength
```

```
$ rubocop test.rb --only Lint/CopDirectiveSyntax
Inspecting 1 file
W

Offenses:

test.rb:1:7: W: Lint/CopDirectiveSyntax: Malformed directive comment detected. Cop names must be separated by commas. Comment in the directive must start with --.
a = 1 # rubocop:disable Layout/ LineLength
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

I'm not really sure what the correct fix here should be. The `Lint` namespace disable is "correct" as per the current rules of disable directive -- ie. having a malformed comment does not prevent RuboCop from parsing it as well as possible. We could perhaps not apply a disable of the `Lint` department to the `Lint/CopDirectiveSyntax` cop specifically, but that might be surprising for users. 

In any case, I think this should be a separate issue at this point.
